pyteal/compiler/subroutines.py

```diff
@@ -90,6 +117,23 @@ def spillLocalSlotsDuringRecursion(
     """
     recursivePoints = findRecursionPoints(subroutineGraph)
 
```

Comment:
@tzaffi Unrelated to the line:  Have you reviewed https://pyteal.readthedocs.io/en/stable/control_structures.html#recursion to determine if user guide changes are warranted?

I still don't have a great feel for how extensively we ought to cover topics in the user guide.  If you've reviewed and feel no change is warranted, I can be on-board.  Minimally, I'm gauging that user guide updates are given consideration.

Reply:
Good call out. I commit to modifying the docs in this PR. Will leave this unresolved as a "visual reminder"

Reply:
I believe this is resolvable, assuming we're good with the following docs update:
<img width="717" alt="image" src="https://user-images.githubusercontent.com/291133/161111050-1a977a74-669f-4835-b345-347b884c73aa.png">